Index: src/system/spare/solving/complete_info/complete_customer_info/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/system/spare/solving/complete_info/complete_customer_info/index.vue	(revision d8525f6b4e298b1e82c33c9dff318ab1cc4cc261)
+++ src/system/spare/solving/complete_info/complete_customer_info/index.vue	(revision )
@@ -49,7 +49,7 @@
               <el-input v-model="customerAddInfo.workPhoneNum" placeholder="请填写单位固话" :disabled="hasSubmit"></el-input>
             </el-form-item>
             <el-form-item label="人事部联系人" prop="hrContactName" class="el-form-item__33">
-              <el-input v-model="customerAddInfo.hrContactName" placeholder="请填写人事部联系人" :disabled="hasSubmit"></el-input>
+              <el-input v-model="customerAddInfo.hrContactName" placeholder="请填写人事部联系人" maxlength="30" :disabled="hasSubmit"></el-input>
             </el-form-item>
             <el-form-item label="人事部联系人手机号码" prop="hrContactNum" class="el-form-item__33">
               <el-input v-model="customerAddInfo.hrContactNum" placeholder="请填写人事部联系人手机号" :disabled="hasSubmit"></el-input>
@@ -112,6 +112,7 @@
                      class="demo-form-inline" label-position="top" >
               <el-form-item label="亲属联系人" maxlength="15" prop="name" class="el-form-item__25">
                 <el-input
+                  maxlength="15"
                   v-model="customerRelation.name"
                   :disabled="(this.customerContactsInfo.maritalStatusCode === '02' || this.customerContactsInfo.maritalStatusCode === '05') || hasSubmit"
                   placeholder="请填写姓名"></el-input>
@@ -126,8 +127,9 @@
                   <el-option label="其他" value="Oth"></el-option>
                 </el-select>
               </el-form-item>
-              <el-form-item label="亲属联系人其他说明" prop="remark" class="el-form-item__25">
+              <el-form-item  label="亲属联系人其他说明" prop="remark" class="el-form-item__25">
                 <el-input
+                  maxlength="10"
                   :disabled="(this.customerContactsInfo.maritalStatusCode === '02' || this.customerContactsInfo.maritalStatusCode === '05') || (customerRelation.relationshipCode !== 'Oth') || hasSubmit"
                   v-model="customerRelation.remark"
                   placeholder="请填写其他说明"></el-input>
@@ -146,7 +148,7 @@
                      :disabled="hasSubmit"
                      class="demo-form-inline" label-position="top" >
               <el-form-item label="非亲属联系人" prop="name" class="el-form-item__25">
-                <el-input v-model="customerNonRelation.name" placeholder="请填写姓名" :disabled="hasSubmit"></el-input>
+                <el-input maxlength="15" v-model="customerNonRelation.name" placeholder="请填写姓名" :disabled="hasSubmit"></el-input>
               </el-form-item>
               <el-form-item label="非亲属联系人关系" prop="relationshipCode" class="el-form-item__25">
                 <el-select
@@ -161,6 +163,7 @@
               </el-form-item>
               <el-form-item label="其他说明" prop="remark" class="el-form-item__25">
                 <el-input
+                  maxlength="10"
                   v-model="customerNonRelation.remark"
                   :disabled="(customerNonRelation.relationshipCode !== 'Oth') || hasSubmit"
                   placeholder="请填写其他说明"></el-input>
@@ -247,7 +250,7 @@
               <button class="button"
                       type="button"
                       :disabled="hasSubmit"
-                      @click="dropSelectShow1=!dropSelectShow1"
+                      @click="dropSelectShow1=!dropSelectShow1;dropSelectShow2 = false;dropSelectShow3 = false;"
                       :class="{isactive:dropSelectShow1}">{{selectChannel.channelName}} <i class="icon el-select__caret el-input__icon el-icon-arrow-up"></i> </button>
               <transition name="el-zoom-in-top">
                 <div class="wrap" v-show="dropSelectShow1">
@@ -265,7 +268,7 @@
               <p class="lab">其他渠道说明</p>
               <button class="button"
                       type="button"
-                      @click="dropSelectShow2=!dropSelectShow2"
+                      @click="dropSelectShow2=!dropSelectShow2;dropSelectShow1 = false;dropSelectShow3 = false;"
                       :class="{isactive:dropSelectShow2, disabled: !(selectChannel.exclusiveChannelId === '000001') || hasSubmit}"
                       :disabled="!(selectChannel.exclusiveChannelId === '000001')|| hasSubmit"> {{selectOtherChannel.channelName}} <i class="icon el-select__caret el-input__icon el-icon-arrow-up"></i> </button>
               <transition name="el-zoom-in-top">
@@ -354,7 +357,7 @@
               <button class="button"
                       type="button"
                       :disabled="hasSubmit"
-                      @click="dropSelectShow3=!dropSelectShow3"
+                      @click="dropSelectShow3=!dropSelectShow3;dropSelectShow1 = false;dropSelectShow2 = false;"
                       :class="{isactive:dropSelectShow3}">{{selectManagerChannel.userName}}<i class="icon el-select__caret el-input__icon el-icon-arrow-up"></i> </button>
               <transition name="el-zoom-in-top">
                 <div class="wrap table__wrap-min" v-show="dropSelectShow3">
@@ -390,7 +393,7 @@
               <el-input v-model="marketingInfo.assistantId" placeholder="请填写业务助理" disabled></el-input>
             </el-form-item>
             <el-form-item label="授权原因" class="el-form-item__25" >
-              <el-select v-model="marketingInfo.authReasonCode" placeholder="请选择亲授权原因" disabled>
+              <el-select v-model="marketingInfo.authReasonCode" placeholder="请选择授权原因" disabled>
                 <el-option label="业务正常" value="N"></el-option>
                 <el-option label="否决重报业务" value="R"></el-option>
               </el-select>
@@ -481,6 +484,7 @@
             </el-form-item>
             <el-form-item label="其他贷款用途说明" prop="loanOtherPurpose" class="el-form-item__25"  >
               <el-input
+                maxlength="10"
                 :disabled="!(orgSpecialInfo1.loanPurposeCode === 'Oth') || hasSubmit"
                 v-model="orgSpecialInfo1.loanOtherPurpose"
                 placeholder="请填写其他贷款用途说明"></el-input>
@@ -565,6 +569,7 @@
             </el-form-item>
             <el-form-item label="其他贷款用途说明" prop="loanOtherPurposeD" class="el-form-item__25" >
               <el-input
+                maxlength="10"
                 v-model="orgSpecialInfo2.loanOtherPurpose"
                 :disabled="!(orgSpecialInfo2.loanPurposeCode === 'Oth') || hasSubmit" placeholder="请填写其他贷款用途说明"></el-input>
             </el-form-item>
@@ -1475,15 +1480,21 @@
         }
       },
       checkloanOtherPurpose1(rule, value, callback) {
+        var reg = /[\u4E00-\u9FA5]+/;
         if ((this.orgSpecialInfo1.loanPurposeType === 'S') && (this.orgSpecialInfo1.loanPurposeCode === 'Oth') && !value) {
           callback(new Error('请填写其他贷款用途说明'));
+        } else if ((this.orgSpecialInfo1.loanPurposeType === 'S') && (this.orgSpecialInfo1.loanPurposeCode === 'Oth') && !reg.test(value)) {
+          callback(new Error('只能输入汉字'));
         } else {
           callback();
         }
       },
       checkloanOtherPurpose2(rule, value, callback) {
-        if ((this.orgSpecialInfo2.loanPurposeType === 'S') && (this.orgSpecialInfo1.loanPurposeCode === 'Oth') && !value) {
+        var reg = /[\u4E00-\u9FA5]+/;
+        if ((this.orgSpecialInfo2.loanPurposeType === 'S') && (this.orgSpecialInfo2.loanPurposeCode === 'Oth') && !value) {
           callback(new Error('请填写其他贷款用途说明'));
+        } else if ((this.orgSpecialInfo2.loanPurposeType === 'S') && (this.orgSpecialInfo2.loanPurposeCode === 'Oth') && !reg.test(value)) {
+          callback(new Error('只能输入汉字'));
         } else {
           callback();
         }
